front_end/content.py

```diff
@@ -636,50 +676,65 @@ def get_course_scores(self, course_id):
     def get_assignment_scores(self, course_id, assignment_id):
         scores = []
 
-        sql = '''SELECT user_id, (SUM(score) / b.num_problems) AS percent_passed
-                 FROM scores
+        sql = '''SELECT u.name, s.user_id, (SUM(s.score) / b.num_problems) AS percent_passed
+                 FROM scores s
+                 INNER JOIN users u
+                 ON s.user_id = u.user_id
                  INNER JOIN (
                    SELECT COUNT(DISTINCT problem_id) AS num_problems
                    FROM problems
                    WHERE course_id = ?
                      AND assignment_id = ?
                      AND visible = 1
                   ) b
-                 WHERE course_id = ?
-                  AND assignment_id = ?
-                  AND user_id NOT IN
+                 WHERE s.course_id = ?
+                  AND s.assignment_id = ?
+                  AND s.user_id NOT IN
                    (
                     SELECT user_id
                     FROM permissions
-                    WHERE course_id = 0 OR course_id IS NULL
+                    WHERE course_id = 0 OR course_id = ?
                    )
-                 GROUP BY course_id, assignment_id, user_id'''
-
-        self.cursor.execute(sql, (int(course_id), int(assignment_id), int(course_id), int(assignment_id),))
+                  AND s.problem_id NOT IN
+				   (
+				    SELECT problem_id
+					FROM problems
+					WHERE course_id = ?
+					AND assignment_id = ?
+					AND visible = 0
+				   )
+                 GROUP BY s.course_id, s.assignment_id, s.user_id'''
+
+        self.cursor.execute(sql, (int(course_id), int(assignment_id), int(course_id), int(assignment_id), int(course_id), int(course_id), int(assignment_id),))
 
         for user in self.cursor.fetchall():
-            scores_dict = {"user_id": user["user_id"], "percent_passed": user["percent_passed"]}
+            scores_dict = {"name": user["name"], "user_id": user["user_id"], "percent_passed": user["percent_passed"]}
             scores.append([user["user_id"], scores_dict])
 
         return scores
 
     def get_problem_scores(self, course_id, assignment_id, problem_id):
         scores = []
 
```

Comment:
Does this query need to use the submissions table? Could it not use the scores table (only) to make it faster and perhaps simpler?

Reply:
It needs to use the submissions table to get the total number of submissions for each problem. The scores table only stores one score for each problem.